Port to Plasma 6

felagund opened the pull request:
Hello,
this is a finish of the port to Plasma 6, now everything works.

felagund committed:
Update CMakeLists.txt to build and install correctly on Fedora 42

felagund committed:
Make synced icon appear

felagund committed:
Update README.md

felagund committed:
Update README.md – Fix links to docs

felagund committed:
Make menu action work on Plasma 6 (and remove obsolete things)

felagund committed:
Rename files so they adhere to KDE naming conventions

Look at your /usr/lib64/qt6/plugins/kf6/kfileitemaction/
(or lib if you are on Debian), first is the name of the app
and then "fileitemaction"

felagund committed:
Rename also inside files now that they were renamed

felagund committed:
Add and update license information

felagund committed:
Fix menu item being shown for files that are being uploaded

This would have been fix for
https://github.com/insynchq/dolphin-insync-plugin/issues/3
https://bugs.kde.org/show_bug.cgi?id=439664
https://forums.insynchq.com/t/insynch-crashing-dolphin/18686

With Qt 6, this was not crashing anymore, it was just showing useless
"n" menu item.

The problem was that the code was checking whether responce was null,
but in fact it was a QByteArray with value "null".

guiodic commented:
@felagund too much now...
<img width="258" height="231" alt="Screenshot_2025-10-08_12 08 31" src="https://github.com/user-attachments/assets/2a094b2d-0ab7-4eaa-a1fb-80705cafaf94" />

felagund committed:
Add icons and fix readme

Add emblems to installation so that they are displayed,
also updated readme to reflect that and update link to Qt documentation.

felagund commented:
That is because I renamed the files to match KDE conventions and now you have the plugin installed twice.  Run

```
sudo rm /usr/lib64/libinsyncdolphinpluginhelper.so /usr/lib64/qt6/plugins/kf6/kfileitemaction/fileitemactioninsyncplugin.so /usr/lib64/qt6/plugins/kf6/overlayicon/overlayiconinsyncplugin.so
/usr/share/kio/servicemenus/fileitemactioninsyncplugin.desktop 
```
and that should fix that, I hope. In the last commit, I am also including the icons you posted - you might want to remove the ones installed from the deb.

guiodic commented:
@felagund yeah, I had imagined something like that, but removing the old copy works fine.

felagund committed:
Update README.md - specify KDE version

felagund committed:
Add icon to json file for fileitemaction

This causes the icon to be visible in Dolphin preferences.

felagund committed:
Remove old files during install

After renaming the files in commit 1ed1b8f11c72d020043ada1dfdb80eff28c52476,
emblems would appear twice if the previous files were still present,
so let's remove them.

felagund committed:
Remove one uninstall line that caused failed install

felagund committed:
Update README.md - update Build instructions to use one command less

felagund committed:
Fix Dolphin crash when Insync was not running or when it was starting

felagund committed:
Readme: Add dependencies for Fedora installation

felagund committed:
Update README with dependencies for Kubuntu

Added installation instructions for Kubuntu and related systems.

Fixes https://github.com/felagund/dolphin-insync-plugin-plasma-6/issues/3

felagund committed:
Modify build instructions and installation notes

Updated installation instructions and clarified the installation path.

felagund committed:
Update README with details on forked Insync plugin

felagund committed:
Update README with build and installation instructions

Added build instructions and repository cloning steps.
